src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,134 @@ public function getTableRow(Element $element, $search)
+    $rows = $element->findAll('css', 'tr');
+    return empty($rows);
+  }
+
   /**
    * Find text in a table row containing given text.
    *
    * @Then I should see (the text ):text in the :rowText row
    */
-    public function assertTextInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) === false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    public function assertTextInTableRow($text, $rowText) {
+      $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+      foreach ($rows as $row) {
+        if (str_contains($row->getText(), $text) !== false) {
+          return;
         }
+      }
+      throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    }
+
+  /**
+   * Find text in all table rows containing given text.
+   *
+   * @Then I should see (the text ):text in all the :rowText rows
+   */
+  public function assertTextInTableRows($text, $rowText) {
+    $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+    foreach ($rows as $row) {
+      if (str_contains($row->getText(), $text) !== false) {
+        throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+      }
     }
+  }
 
   /**
-   * Asset text not in a table row containing given text.
+   * Assert text not in a table row containing given text.
+   * If multiple rows are found, check the first one only.
    *
    * @Then I should not see (the text ):text in the :rowText row
    */
-    public function assertTextNotInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) !== false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+  public function assertTextNotInTableRow($text, $rowText)
+  {
+    $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
+    if (strpos($row->getText(), $text) !== false) {
+      throw new \Exception(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+    }
+  }
+
+  /**
+   * Asset text not in any table rows containing given text.
+   *
+   * @Then I should not see (the text ):text in any :rowText rows
+   */
+    public function assertTextNotInAnyTableRows($text, $rowText): void {
+        $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+        foreach ($rows as $row) {
+          if (strpos($row->getText(), $text) !== false) {
+            throw new \RuntimeException(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+          }
         }
```

Comment:
@kumikoono Let's use a guard clause:

```suggestion
        foreach ($rows ?: [] as $row) {
          if (strpos($row->getText(), $text) === FALSE) {
            continue;
          }
          throw new \RuntimeException(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
        }
```